src/Control/CookieJar.php

```diff
@@ -180,4 +190,20 @@ protected function outputCookie(
         }
         return false;
     }
+
+    /**
+     * Get the correct samesite value - if this is a session cookie it may be different to the default value.
+     *
+     * @deprecated 5.0 The relevant methods will include a `$sameSite` parameter instead.
+     */
+    private function getSameSite(string $name): string
+    {
+        if ($name === session_name()) {
+            return Session::config()->get('cookies_samesite') ?: Cookie::config()->get('default_samesite');
```

Comment:
This logic is pretty confusing, I've made a comment later on about consolidating this config